include/zephyr/drivers/spi.h

```diff
@@ -480,6 +482,16 @@ typedef int (*spi_api_io_async)(const struct device *dev,
 				spi_callback_t cb,
 				void *userdata);
 
+#ifdef CONFIG_SPI_RTIO
```

Comment:
I think you need `#if defined(CONFIG_SPI_RTIO) || defined(DOXYGEN)` for the docs to generate correctly